pkg/ingester/ingester_ring.go

```diff
@@ -69,6 +69,9 @@ func (cfg *RingConfig) Validate() error {
 	}
 
 	if cfg.TokenGenerationStrategy == spreadMinimizingTokenGeneration {
+		if cfg.TokensFilePath != "" {
+			return fmt.Errorf("bad configuration: %q token generation strategy requires \"tokens-file-path\" to be empty", spreadMinimizingTokenGeneration)
```

Comment:
[nit] A couple of small things:
1. No need to say "bad configuration". The validation errors are decorated by the caller.
2. Do not hardcode "tokens-file-path" but read it from a const shared with `RegisterFlags()`